lib/internal/fs/promises.js

```diff
@@ -210,9 +218,31 @@ class FileHandle extends JSTransferable {
   [kDeserialize]({ handle }) {
     this[kHandle] = handle;
     this[kFd] = handle.fd;
+    this[kEmitter] = new EventEmitter();
+  }
+
+  [kRef]() {
+    this[kRefs]++;
+  }
+
+  [kUnref]() {
+    this[kRefs]--;
+    if (this[kRefs] === 0) {
+      this[kFd] = -1;
+      this[kHandle]
+        .close()
+        .then(this[kCloseResolve], this[kCloseReject]);
+    }
   }
 }
 
+for (const op of ['on', 'emit', 'once', 'off', 'addListener',
+                  'prependListener', 'removeListener', 'removeAllListeners',
+                  'listeners', 'rawListeners'])
+  FileHandle.prototype[op] = function(...args) {
+    this[kEmitter][op].apply(this[kEmitter], args);
```

Comment:
```suggestion
    ReflectApply(this[kEmitter][op], this[kEmitter], args);
```

`ReflectApply` is available from the `primordials` object, at the top of the file.